lib/amazon-pricing/definitions/database-type.rb

```diff
@@ -232,6 +232,23 @@ def self.display_name_is_byol?(display_name)
       database_name.include? 'byol'
     end
 
+    # example: database_sf_from_engine_name?('mysql', false, false) returns true
+    # Returns BOOL if database is RDS SF
+    # params:
+    # - engine_name[String]: product description database name string
+    # - is_byol[Bool]: true if the database is using a BYOL license
+    # - is_multi_az[Bool]: true is the deployment type is Multi-AZ
+    def self.database_sf_from_engine_name?(engine_name, is_byol, is_multi_az)
```

Comment:
this function seems to be a superset of `self.db_mapping` method.

for example, we can fix AWS-1559 by changing clients using @@ProductDescription  to ignore extraneous space before '(byol)' and '(li)' (without changing the that hash);  hence change db_mapping to handle these cases, and perhaps you can do the same?

Reply:
I'm not sure I understand the suggestion - is this what you were thinking of? https://github.com/CloudHealth/amazon-pricing/pull/141/commits/4e062a0420fd9825c6c9520a26eb7d8705709b54